### DISCUSSION TITLE
[Custom]: Wallust not changing colorscheme for some wallpapers (mostly monochrome wallpapers).

### DISCUSSION THREAD

**Discussion by Yaaneva** [JaKooLit/Hyprland-Dots#450] [General]
### Description

Wallust does work for most of the vibrant wallpapers but fails to change the colorscheme when the wallpaper is somewhat monochrome (it also doesn't work for some colorful wallpapers too). After changing the value of threshold to 2 in wallust.toml it does recognize some wallpapers that it didn't previously recognize. But it still remains the same for monochrome wallapapers as shown in the video below. The waybar and rofi color remains the same. I don't know if this is a bug but I haven't seen it occur anywhere else in the internet.

### Expected Behavior

It should work normally for monochrome wallpapers too, I believe.

### Actual Behavior

The colorscheme remains the same as the previous wallpaper and doesn't change for monochrome wallpapers.

### Screenrecording

https://github.com/user-attachments/assets/74730566-ca24-4055-b8bc-f627440ff29e

### Proposed solution

Use fallback colors for monochrome wallpaper, I guess.
For eg:- fallback_colors = ["#FFFFFF", "#000000", "#D3D3D3", "#808080"]

### Additional Information

I did have problems with wallust before too. At the start it didn't change colorscheme at all. That issue has somehow been resolved though. Now it atleast works for most part.

**Comment by JaKooLit** [comment]
I have actually found out about this the other day and I have tried matugen as color-palette..

However, matugen is worst that wallust.

BUT,

I have recently found out by changing some settings on ~/.config/wallust/wallust.toml

will also process the monochrome wallpapers

color_space = "labmixed" 

gives a fairly good colors

you can also play around with backend

**Reply by Yaaneva** [reply]
Changing "lch" to "labmixed" worked like a charm. I would suggest you to change it to that in the next update, I guess.

**Reply by JaKooLit** [reply]
> Changing "lch" to "labmixed" worked like a charm. I would suggest you to change it to that in the next update, I guess.

I have it already

**Comment by Yaaneva** [comment]
> I have actually found out about this the other day and I have tried matugen as color-palette..
> 
> However, matugen is worst that wallust.
> 
> BUT,
> 
> I have recently found out by changing some settings on ~/.config/wallust/wallust.toml
> 
> will also process the monochrome wallpapers
> 
> color_space = "labmixed"
> 
> gives a fairly good colors
> 
> you can also play around with backend

I see. I will do that then. Thank you.
